### PR TITLE
task/internal: account for locked machines acquired on previous loops

### PR DESCRIPTION
**Pull request by dmick**
Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by andrewschoen** [comment]
LGTM

File: teuthology/task/internal.py [hunk 1]
```diff
@@ -159,6 +159,9 @@ def lock_machines(ctx, config):
             break
         elif not ctx.block:
             assert 0, 'not enough machines are available'
+        else:
+            how_many = how_many - len(newly_locked)
+            assert how_many > 0
```
Comment (andrewschoen) [inline teuthology/task/internal.py]: Is this needed?  If it is, can we add a message onto it explaining why the assertion failed?

Comment (zmc) [inline teuthology/task/internal.py]: Agreed; no bare asserts!

Comment (dmick) [inline teuthology/task/internal.py]: ahhh, ya bluenoses.  Fine.  FINE.

Comment (dmick) [inline teuthology/task/internal.py]: (the only reason the assert is there is that I could think of no good way to properly test this without inventing a lot more mechanism.)

**Pull request merged**